### PR TITLE
Implement class decorators

### PR DESCRIPTION
**Pull request by collinsmezie**
Turn Person class to Nameable using inheritance
Prepare base Decorator
Create capitalize decorator 
Create trimmer decorator

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require_relative './nameable'
+
+# base decorator
+class BaseDecorator < Nameable
```
Comment (Deepakdanger) [inline base_decorator.rb:4]: - [ ] Here kindly include `attr_accessor :nameable_obj` here.

File: check_result.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+require_relative './TrimmerDecorator'
+
+person = Person.new(22, 'maximilianus')
+person.correct_name
+capitalized_person = CapitalizeDecorator.new(person)
+capitalized_person.correct_name
+capitalized_trimmed_person = TrimmerDecorator.new(capitalizedPerson)
+capitalized_trimmed_person.correct_name
+
+p person.correct_name
```
Comment (Deepakdanger) [inline check_result.rb]: - [ ] Kindly remove this code, its not needed here.

File: nameable.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Base class
+class Nameable
+  def correct_name
+    raise NotImplementedError, ' name is not correct'
```
Comment (Deepakdanger) [inline nameable.rb]: - [ ] while raising an error the error shown to user should be specific, Kindly use this one 
- `raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"`

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @collinsmezie, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Chime, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**